src/services/projects.js

```diff
@@ -64,10 +86,42 @@ angular.module('openshiftCommonServices')
                               });
                     });
           },
+
+          // List the projects the user has access to. This method returns
+          // cached data if the projects had previously been fetched to avoid
+          // requesting them again and again, which is a problem for admins who
+          // might have hundreds or more.
+          list: function(forceRefresh) {
+            if (cachedProjectData && !forceRefresh) {
+              Logger.debug("ProjectsService: returning cached project data");
+              return $q.when(cachedProjectData);
+            }
+
+            Logger.debug("ProjectsService: listing projects, force refresh", forceRefresh);
+            return DataService.list('projects', {}).then(function(projectData) {
+              cachedProjectData = projectData;
+              return projectData;
```

Comment:
Should this watch for updates to the projects so that future calls get any updates that have occurred since the last retrieval?